src/InputGroup/_InputGroupAddon.js

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+import Icon from '../Icon/Icon';
+import { INPUT_GROUP_ADDON_TYPES } from '../utils/constants';
+import PropTypes from 'prop-types';
+import React from 'react';
+import withStyles from '../utils/WithStyles/WithStyles';
+
+const InputGroupAddon = ({
+    addon,
+    actions,
+    addonClassNames,
+    children,
+    className,
+    compact,
+    disableStyles,
+    glyph,
+    ...otherProps
+}) => {
+
+    let returnObject = children;
+
+    if (glyph && addon === 'icon') {
+        returnObject = (
+            <Icon
+                disableStyles={disableStyles}
+                glyph={glyph}
+                role='presentation' />
```

Comment:
This _seems_ wrong.  I think the consumer should own what `Icon` or other children are within the `InputGroupAddon`.  The only "extra" thing we should need is knowing when the child is a button (or children are buttons).